kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java

```
@@ -0,0 +1,163 @@
+        originalDataSourceDef.setConnectionURL( "connectionURLOriginal" );
+        originalDataSourceDef.setUser( "userOriginal" );
+        originalDataSourceDef.setPassword( "passwordOriginal" );
+    }
+
+    @Override
+    protected DefEditorContent getExpectedContent() {
+        return dataSourceDefEditorContent;
+    }
+
+    @Override
+    protected String getExpectedDefString() {
+        return DataSourceDefSerializer.serialize( dataSourceDef );
+    }
+
+    @Override
+    protected String getExpectedFileName() {
+        return dataSourceDef.getName() + ".datasource";
+    }
+
+    @Override
+    protected Def getExpectedDef() {
+        return dataSourceDef;
+    }
+
+    @Override
+    protected Def getOriginalDef() {
+        return originalDataSourceDef;
+    }
+
+    @Override
+    protected String getOriginalDefString() {
+        return DataSourceDefSerializer.serialize( originalDataSourceDef );
+    }
+
+    @Override
+    protected void verifyCreateConditions( boolean global ) {
+        //we wants that:
+        try {
+            // 2) the definition was deployed
+            verify( runtimeManager, times( 1 ) ).deployDataSource( dataSourceDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
+        }
+        // 3) the notification was fired.
+        NewDataSourceEvent expectedEvent;
+        if ( global ) {
+            expectedEvent = new NewDataSourceEvent( dataSourceDef, SESSION_ID, IDENTITY );
+        } else {
+            expectedEvent = new NewDataSourceEvent( dataSourceDef, project, SESSION_ID, IDENTITY );
+        }
+        verify( newDataSourceEvent, times( 1 ) ).fire( expectedEvent );
+    }
+
+    @Override
+    protected void verifySaveConditions() {
+        //we wants that
+        try {
+            // 3) the definition was deployed
+            verify( runtimeManager, times( 1 ) ).deployDataSource( dataSourceDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
+        }
```

Same comment about swallowing the exception.